client/layout/index.jsx

```diff
@@ -134,6 +139,9 @@ class Layout extends Component {
 	}
 
 	renderMasterbar() {
+		if ( this.props.masterbarIsHidden ) {
+			return null;
+		}
```

Comment:
Could we continue to render the masterbar and hide it with CSS here? Instead of not rendering it at all?

I'm afraid that not rendering it at all could break some flows. For example, some code may rely on `document.getElementById( 'header' )` returning a valid element. I recently fixed one such bug in https://github.com/Automattic/wp-calypso/pull/48046.

Various sections often tweak the layout in rather messy ways, that's why I think this change is risky.

Reply:
The general rationale behind not rendering it was to not load unnecessarily all of the additional components that come together with the masterbar if they will be loaded in vain. To prevent such issues perhaps we could render a placeholder masterbar with the `id="header"`?

Reply:
I see, then some `<EmptyMasterbar />` component that just renders the element would be good.

Reply:
Thanks for the suggestion I added an EmptyMasterbar component. Let me know what you think?